### PR TITLE
Login Component

### PR DESCRIPTION
**Pull request by PeaWarrior**
**Desktop View**
desktop
desktop-error
**Mobile View**
mobile
mobile-error

### CODE REVIEW

**Comment by PeaWarrior** [comment]
Tests
Screen Shot 2020-12-05 at 3 47 55 PM

File: src/components/AccountForms/LoginForm/index.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+} from '@material-ui/core';
+import { Visibility, VisibilityOff, Error } from '@material-ui/icons';
+import { ReactComponent as LoginHero } from './assets/LoginHero.svg';
+import { ReactComponent as TwitterLogo } from './assets/TwitterLogo.svg';
+
+const FORM_TITLE = 'Welcome back!';
+
+const emailRequirements = {
+  required: true,
+  pattern: /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
```
Comment (cfernando415) [inline src/components/AccountForms/LoginForm/index.js:25]: ### Change to:

`required: 'Email is required',
  pattern: {
    value: /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
    message: 'Invalid email format',
  },`

File: src/components/AccountForms/LoginForm/index.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  Paper,
+  TextField,
+  InputAdornment,
+  IconButton,
+  Button,
+} from '@material-ui/core';
+import { Visibility, VisibilityOff, Error } from '@material-ui/icons';
+import { ReactComponent as LoginHero } from './assets/LoginHero.svg';
+import { ReactComponent as TwitterLogo } from './assets/TwitterLogo.svg';
+
```
Comment (cfernando415) [inline src/components/AccountForms/LoginForm/index.js:20]: ### Changes:

`import isEmpty from 'lodash/isEmpty';`

File: src/components/AccountForms/LoginForm/index.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  const { auth } = useAuth();
+
+  useEffect(() => {
+    if (auth.isLoggedIn()) {
+      history.replace('/mentors');
+    }
+  }, [auth, history]);
+
+  const onSubmit = (credentials) => {
+    request.post(API_PATH.LOGIN, credentials).then((response) => {
```
Comment (cfernando415) [inline src/components/AccountForms/LoginForm/index.js:48]: ### Changes:
`then(({ status })) =>`

File: src/components/AccountForms/LoginForm/index.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+      if (response.status === 200) {
```
Comment (cfernando415) [inline src/components/AccountForms/LoginForm/index.js:49]: `if(status === 200)`

File: src/components/AccountForms/LoginForm/index.js [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+                    </IconButton>
+                  </InputAdornment>
+                ),
+              }}
+              FormHelperTextProps={{
+                classes: {
+                  root: classes.helperText,
+                },
+              }}
+            />
```
Comment (cfernando415) [inline src/components/AccountForms/LoginForm/index.js:146]: same es6 changes

File: src/components/AccountForms/LoginForm/index.js [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    <div className={classes.root}>
+      <Paper classes={{ root: classes.paperContainer }}>
+        <LoginHero className={classes.heroImage} />
+        <div className={classes.loginFormContainer}>
+          {(errors.email || errors.password) && (
+            <div className={classes.error} data-testid="authError">
+              <Error />
+              <p role="alert">Sorry, invalid email or password. Try again?</p>
+            </div>
+          )}
```
Comment (cfernando415) [inline src/components/AccountForms/LoginForm/index.js]: ### Changes:
we also want the error to be driven by query params and show the session timeout if the url has timedOut=true, for example https://femmecubator.com/login?timedOut=true

**Timed Out Session**
image

**Review by cfernando415 — CHANGES_REQUESTED** [review]
please changes requested

File: src/components/AccountForms/LoginForm/index.js [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+            <TextField
+              {...{
+                id: 'email',
+                label: 'Email',
+                variant: 'outlined',
+                className: classes.loginInput,
+                inputRef: register(emailRequirements),
+                name: 'email',
+                autoComplete: 'email',
+                error: !isEmpty(errors.email) && true,
```
Comment (cfernando415) [inline src/components/AccountForms/LoginForm/index.js:100]: remove true !isEmpty(errors.email) will resolve to true if errors.email is empty

File: src/components/AccountForms/LoginForm/index.js [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+              {...{
+                id: 'password',
+                label: 'Password',
+                variant: 'outlined',
+                className: classes.loginInput,
+                inputRef: register({ required: true }),
+                name: 'password',
+                autoComplete: 'current-password',
+                type: showPassword ? 'text' : 'password',
+                error: !isEmpty(errors.password) && true,
```
Comment (cfernando415) [inline src/components/AccountForms/LoginForm/index.js:130]: same comment as previous one

File: src/components/AccountForms/LoginForm/index.js [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+                  inputProps: { 'data-testid': 'password' },
+                  classes: {
+                    input: classes.input,
+                  },
+                  endAdornment: (
+                    <InputAdornment position="end">
+                      <IconButton
+                        aria-label="toggle password visibility"
+                        edge="end"
+                        onClick={handleClickShowPassword}
```
Comment (cfernando415) [inline src/components/AccountForms/LoginForm/index.js:148]: `{...{ 'aria-label': 'toggle password visibility', edge: 'end', onClick: handleClickShowPassword }}`

**Review by cfernando415 — CHANGES_REQUESTED** [review]
almost there

File: src/components/AccountForms/LoginForm/index.js [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+      <Paper classes={{ root: classes.paperContainer }}>
+        <LoginHero className={classes.heroImage} />
+        <div className={classes.loginFormContainer}>
+          {!isEmpty(errors) && !isMobile && (
+            <div className={`${classes.alert} ${classes.error}`}>
+              <Error />
+              <p role="alert">Sorry, invalid email or password. Try again?</p>
+            </div>
+          )}
+          {isEmpty(errors) && search === '?timedOut=true' && (
```
Comment (cfernando415) [inline src/components/AccountForms/LoginForm/index.js:80]: ### Change requested:
- used the extracted value of the query string here

`{isEmpty(errors) && timedOut && (`     ....

**Review by cfernando415 — CHANGES_REQUESTED** [review]
please see comment

**Review by cfernando415 — APPROVED** [review]
👍

**Pull request merged**